.github/workflows/cf-deploy.yml

```
@@ -0,0 +1,77 @@
+name: Deploy to bucket
+
+on:
+  push:
+    branches: ['DOI-1529-switchboard-cloudfront-ci-cd']
+jobs:
+  cancel-previous:
+    name: 'Cancel Previous Runs'
+    runs-on: ubuntu-latest
+    timeout-minutes: 3
+    steps:
+      - uses: styfle/cancel-workflow-action@0.8.0
+        with:
+          access_token: ${{ github.token }}
+  env_name:
+    runs-on: ubuntu-latest
+    steps:
+      - name: Get GHA environment name
+        id: env_name
+        run: |
+          echo "Running on branch ${{ github.ref }}"
+          if [ "${{ github.ref }}" = "refs/heads/DOI-1529-switchboard-cloudfront-ci-cd" ]; then
+            echo "::set-output name=env_name::sandbox"
+          fi
+      - name: Use specified environment secrets from previous step
+        run: echo "Environment secrets set name ${{ steps.env_name.outputs.env_name }}"
+    outputs:
+      env_name: ${{ steps.env_name.outputs.env_name }}
+  deploy:
+    needs: [cancel-previous, env_name]
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v2
+        with:
+          fetch-depth: 0
+      - uses: actions/setup-node@v2-beta
+        with:
+          node-version: '10.22.0'
+          registry-url: https://registry.npmjs.org/
+      - name: Cache node modules
+        uses: actions/cache@v2
+        env:
+          cache-name: cache-node-modules
+        with:
+          path: ~/.npm
+          key: ${{ runner.os }}-build-${{ env.cache-name }}-${{ hashFiles('**/package-lock.json') }}
+          restore-keys: |
+            ${{ runner.os }}-build-${{ env.cache-name }}-
+            ${{ runner.os }}-build-
+            ${{ runner.os }}-node-
+            ${{ runner.os }}-
+      - name: Install npm packages
+        run: |
+          npm install
+          envsubst < src/environments/constants.template.ts > src/environments/constants.ts
+      - name: Build master branch
+        run: |
+          cp src/environments/environment.dev.ts src/environments/environment.ts
+          npm run build
+      - name: Ensure dist is present
+        run: |
+          test -d "dist" && echo OK || exit 1
+      - name: Configure AWS credentials
+        uses: aws-actions/configure-aws-credentials@v1
+        with:
+          aws-access-key-id: ${{ secrets.AWS_ACCESS_KEY_ID }}
+          aws-secret-access-key: ${{ secrets.AWS_SECRET_ACCESS_KEY }}
+          aws-region: ${{ secrets.AWS_REGION }}
+      - name: Clear bucket
+        run: |
+          aws s3 rm s3://${{ secrets.BUCKET_NAME }} --recursive
+      - name: Upload to bucket
+        run: |
+          aws s3 cp dist s3://${{ secrets.BUCKET_NAME }} --recursive
+      - name: Invalidate CloudFront
```

Why this `Invalidate Cloudfront` step?

---

Yes, I'm taking care of it.

Cloudfront caches s3 content so after each deployment, it needs to be cleared.

---

Thanks, what do you think of adding this as a comment?
```suggestion
      # cloudfront caches s3 content so after each deployment, it needs to be cleared.
      - name: Invalidate CloudFront
```